provider/filepath.go

```diff
@@ -0,0 +1,99 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+package provider
+
+import (
+	"fmt"
+	"net/url"
+	"strings"
+)
+
+type Filepath interface {
+	Bucket() string
+	Prefix() string
+	Path() string
+	FullPathWithBucket() string
+	FullPathWithoutBucket() string
+	ParseFullPath(path string) error
+}
+
+func NewFilepath(storeType string, bucket string, prefix string, path string) (Filepath, error) {
+	switch storeType {
+	case "s3":
```

Comment:
Should be an enum

Reply:
should we make Filestore type as enum as well?